src/thematic/0_generic/sfp-001_vervanging-terrest_nl/07_stappenplan.Rmd

```diff
@@ -0,0 +1,117 @@
+
+#### Werkwijze 1: kartering van doelhabitat binnen polygoon {#werkwijze1}
+
+De eerste werkwijze wordt geïllustreerd in Figuur \@ref(fig:fig2) waarbij alle aanwezige doelhabitat wordt gekarteerd binnen de polygoon. Alle habitatvlekken van het doelhabitat worden dus in kaart gebracht. Bij voorkeur gebeurt deze kartering in de GIS-omgeving op de veldcomputer. Vervolgens kijk je welke potentiële vervangpunten in de gekaartede habitatvlekken vallen en selecteer je het vervangpunt met het laagste rangnummer. In het (fictieve) voorbeeld in Figuur \@ref(fig:fig2) selecteren we vervangpunt met rangnummer 3. Dit is het punt binnen het doelhabitat met het laagste relatieve rangnummer. 
+
+```{r fig2, fig.cap = "De geselecteerde steekproefeenheid (geel) is niet geschikt. Na kartering van het doelhabitat (gearceerde polygoon), wordt het vervangpunt met laagste rangnummer binnen het doelhabitat als vervangpunt geselecteerd"}
+
+knitr::include_graphics("media/vervangingsprocedure_1_stap2_metraster.png")
+
+```
+
+#### Wekwijze 2: evalueer de mogelijke vervangpunten in volgorde volgens rangnummer {#werkwijze2}
+
+Figuur \@ref(fig:fig3) illustreert een tweede mogelijkheid waarbij de potentiële vervangpunten in volgorde worden bezocht tot er een geschikt vervangpunt wordt gevonden. In het (fictieve) voorbeeld in Figuur\@ref(fig:fig3) wandel je dus eerst naar punt 1 (= niet geschikt), vervolgens naar punt 2 (= niet geschikt) en kom je vervolgens bij punt 3 dat wel geschikt blijkt te zijn. Als je onderweg in de buurt van andere punten passeert waarvan je snel kan inschatten dat ze niet in doelhabitat vallen, kan je dit uiteraard ook al noteren, zodat je daar niet meer moet terugkomen. Bijvoorbeeld, als je van punt 1 naar punt 2 wandelt, kan je mogelijks al zien dat punt 4 en 5 in een bosrand liggen. Dan kan je die punten al elimineren als mogelijke vervangpunten. 
+
+Deze werkwijze is minder aangewezen voor polygonen met een grote oppervlakte omdat de kans dan groot is dat je dan heel wat heen en weer moet wandelen. Uiteraard moeten beide mogelijke werwijzes tot het zelfde resultaat leiden.  
+
+```{r fig3, fig.cap = "De geselecteerde steekproefeenheid (geel) is niet geschikt. De potentiële vervangpunten worden in volgorde afgelopen en geëvalueerd tot een geschikt vervangpunt wordt gevonden"}
+
+knitr::include_graphics("media/vervangingsprocedure_1_stap2_bis_metraster.png")
+
+```
+#### Wat als er geen geschikt vervangpunt wordt gevonden?
+
+Wanneer er geen doelhabitat wordt gevonden in de polygoon zal er uiteraard geen vervangpunt kunnen geselecteerd worden. De vervangprocedure wordt afgerond en er wordt dus geen bemonstering uitgevoerd in de polygoon.
+
+Het kan ook voorkomen dat er wel doelhabitat aanwezig is, maar dat geen enkel vervangpunt (centroïde van een rastercel) in doelhabitat gelegen is. Dit komt voor in het (fictieve) voorbeeld in Figuur \@ref(fig:fig4). Ook in dit geval wordt er dus geen bemonstering uitgevoerd in de polygoon.
```

Comment:
> Ook in dit geval wordt er dus geen bemonstering uitgevoerd in de polygoon.

Dit is te vroeg, zie andere opmerking (translatie). M.a.w. als het doelhabitat (bemonsterbaar) voorkomt in de polygoon, dan mag het niet vervallen.

Uit het [rapport van de doelpopulatieverkenning](https://drive.google.com/uc?export=download&id=1jH7rNSHvXycCiCrsoMRLaZ3IlY7GX35D):

> De polygon-constrained vervangmethoden zijn verplicht voor de centroïdgekoppelde terrestrische types, vermits de phab-fractie van de polygoon al gebruikt was voor weging in het celselectieproces en we dus effectief een steekproefeenheid wensen die het type is, om geen vertekening te introduceren in de sampling weights.

Reply:
OK. Translatie toegevoegd, maar nog niet praktisch uitgewerkt.

Reply:
Kan er geen pragmatischere methode bedacht worden? Bv: je selecteert eerste rastercel dat doeltype bevat. Je splits de rastercellen in 4 kwadranten, je genereert een random volgorde van de kwadranten en beoordeelt de centroïden van de kwadranten. Moest je dan in zeldzame gevallen toch nog niets vinden, kies je een punt in het eerste kwadrant dat het doeltype bevat.

Reply:
Je voorstel heeft inderdaad een aantal zaken gemeen met de translatie-methode maar zonder dat je alles moet karteren ofwel telkens een nieuwe puntenronde opstart.

Ik ben ook akkoord dat je de eenheid kunt selecteren ter hoogte van de eerste cel die de eenheid bevat, nadat je eerst alle gewone centroïden hebt beoordeeld, dit is meer praktisch.

Wel deze methode best zo zien dat je geen deel van de polygoon mist met de nieuwe centroïden, want er zijn stukken van de polygoon die _buiten_ de aanvankelijk beschouwde rastercellen vallen (hun centroïd valt buiten polygoon), terwijl daar wel kwadrantcentroïden (of anders bekomen nieuwe punten) in vallen. Een vervangmethode die toelaat om de **volledige polygoon** af te zoeken, is meest kansrijk (en logisch, ervan uitgaand dat het doeltype er ergens moet voorkomen).

- In de praktijk zou je bv. kunnen gaan naar een verdichting tot een 16x16m raster. Diens centroïden komen dan overeen met je kwadrantcentroïden, maar beslaan opnieuw de hele polygoon, dus er komen er nog een aantal zones bij. Het is vanuit zo'n 'kwadrant'gedachte dat die translatiebenadering uiteindelijk gegroeid is trouwens. De uitdaging bij een verdichting is hoe je de bijkomende punten aan de randen, die niet binnen je aanvankelijke eenheden liggen, kunt betrekken in de selectieprocedure. De meest logische aanpak is hier om eveneens het GRTS-rangnummer van de aansluitende cellen (32x32) mee in rekening te nemen zodat je dezelfde aanpak kunt volgen: welk 32x32 cel moet je eerst beschouwen om kwadrantcentroïden te beoordelen?
- Een nadeel van kwadrantcentroïden is dat ze maar `sqrt(128) = 11.3` meter van het oorspronkelijke centroïd liggen, en dat je dus niet halverwege gaat zoeken tussen de eerste set centroïden. Daarom dat we uiteindelijk een translatie verkozen van 16 meter noordwaarts en/of 16 meter oostwaarts (in eerste instantie) - de combinatie van beide komt neer op het bekijken van de hoekpunten van de originele rastercellen. In dit geval moet je ook niet naar punten buiten de originele rastercellen gaan zoeken. Je kijkt dus naar hoekpunten + halverwege de zijden van de reeds beschouwde rastercellen.

Je voorstel, dat in essentie meerdere punten toetst per oorspronkelijke centroïd, heeft wel het nadeel dat je bij de tweede vervang-run niet langer een ruimtelijk gebalanceerde volgorde van nieuwe punten overloopt in de hele polygoon. Maar als je maar één eenheid nodig hebt, lijkt me dat effect te verwaarlozen. Als je in dezelfde polygoon echter twee eenheden nodig hebt, dan is het belangrijk om voor het tweede vervangpunt te starten vanaf de volgende rastercel in de rij waar je **nog geen** sub-punten hebt geëvalueerd.

Voorstel:

1. overloop de centroïden van het originele raster in GRTS-volgorde. De eerste treffer bepaalt de vervangeenheid.
2. levert dit niets op, bekijk dan de hoekpunten + halverwege de zijden van de eerste rastercel die het type _bevat_.
    - De volgorde hiervan kun je randomiseren. Je zou dit ruimtelijk gebalanceerd kunnen doen door een reeks conceptuele GRTS-rasters van 3x3 cellen aan te leggen (telkens één te verbruiken). Conceptueel: enkel N-richting van belang voor toepassing, resolutie maakt niet uit, kan dus op apart blad. Het komt er slechts op aan dat je 9 vakjes met volgordes hebt, die je kunt gebruiken om de 8 randpunten een volgorde toe te kennen.
    - De eerste treffer bepaalt de vervangeenheid.
3. als geen enkel van de randpunten (van de geselecteerde rastercel) het type treffen:
    - selecteer het randpunt (incl hoekpunten dus) dat het dichtst bij het doeltype ligt. Dit ook naar buiten de cel toe beschouwd, tot maximum afstand `sqrt(128) = 11.3` meter, meer specifiek is dit volgens 16x16m cellen met centroïd op rand van originele cel. Reden: de volledige polygoon afdekken! Merk op dat deze kleinere cellen daardoor onderling overlappen (afstand tussen randpunten is 8 meter), maar dit lijkt me geen probleem in deze procedure. Belangrijker is dat ze 16 meter buiten de rastercel reiken, ivm polygoonafdekking.
    - neem een random punt binnen de 16x16m cel van het geselecteerde randpunt. Hiervoor is ook een procedure nodig. Dit punt definieert nu de vervangeenheid (32x32), namelijk als nieuw centroïd ervan.
4. indien nodig voor de uitvoering van het bemonsterings- of meetprotocol, reconstrueer de grenzen van de vervangeenheid (grenzen van de rastercel), met de richting van de celranden conform deze van het originele raster.